subsys/metrics/src/main/java/org/commonjava/indy/metrics/reporter/ReporterIntializer.java

```diff
@@ -0,0 +1,80 @@
+package org.commonjava.indy.metrics.reporter;
+
+import com.codahale.metrics.ConsoleReporter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.graphite.Graphite;
+import com.codahale.metrics.graphite.GraphiteReporter;
+import org.commonjava.indy.metrics.conf.IndyMetricsConfig;
+import org.commonjava.indy.metrics.filter.HealthCheckMetricFilter;
+import org.commonjava.indy.metrics.filter.JVMMetricFilter;
+import org.commonjava.indy.metrics.filter.SimpleMetricFilter;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.io.File;
+import java.io.FileInputStream;
+import java.net.InetSocketAddress;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by xiabai on 3/3/17.
+ */
+@ApplicationScoped
+public class ReporterIntializer {
+
+    @Inject
+    IndyMetricsConfig config;
+
+    public void initReporter(MetricRegistry metrics) throws Exception {
+
+        if (config.getReporter() == null || ("").equals(config.getReporter())) {
```

Comment:
I think this should method body should only execute if the config.isEnabled() == true...